--- Gemfile.lock.orig	2013-01-20 18:52:32.455131334 -0700
+++ Gemfile.lock	2013-01-20 18:46:44.530892708 -0700
@@ -6,6 +6,13 @@
     database_cleaner (0.9.1)
 
 GIT
+  remote: https://github.com/travisghansen/charlock_holmes.git
+  revision: 22ae0b82942c24d35f641567dbbd318d0f0d1c8f
+  ref: 22ae0b82942c24d35f641567dbbd318d0f0d1c8f
+  specs:
+    charlock_holmes (0.6.9)
+
+GIT
   remote: https://github.com/ctran/annotate_models.git
   revision: be4e26825b521f0b2d86b181e2dff89901aa9b1e
   specs:
@@ -129,7 +136,6 @@
     carrierwave (0.7.1)
       activemodel (>= 3.2.0)
       activesupport (>= 3.2.0)
-    charlock_holmes (0.6.9)
     childprocess (0.3.6)
       ffi (~> 1.0, >= 1.0.6)
     chosen-rails (0.9.8)
@@ -461,6 +467,7 @@
   bootstrap-sass (= 2.2.1.1)
   capybara
   carrierwave (~> 0.7.1)
+  charlock_holmes!
   chosen-rails (= 0.9.8)
   coffee-rails (~> 3.2.2)
   colored
--- Gemfile.orig	2013-01-20 18:52:23.303125885 -0700
+++ Gemfile	2013-01-20 18:53:00.944147282 -0700
@@ -1,5 +1,10 @@
 source "http://rubygems.org"
 
+if RUBY_VERSION =~ /1.9/
+  Encoding.default_external = Encoding::UTF_8
+  Encoding.default_internal = Encoding::UTF_8
+end
+
 def darwin_only(require_as)
   RUBY_PLATFORM.include?('darwin') && require_as
 end
@@ -28,6 +33,9 @@
 gem 'grack',         git: "https://github.com/gitlabhq/grack.git",          ref: 'ba46f3b0845c6a09d488ae6abdce6ede37e227e8'
 gem 'grit_ext',      git: "https://github.com/gitlabhq/grit_ext.git",       ref: '8e6afc2da821354774aa4d1ee8a1aa2082f84a3e'
 
+# Patched to fix issues with compiling on Gentoo
+gem "charlock_holmes", git: "https://github.com/travisghansen/charlock_holmes.git", ref: '22ae0b82942c24d35f641567dbbd318d0f0d1c8f'
+
 # Gitolite client (for work with gitolite-admin repo)
 gem "gitolite", '1.1.0'
 
