--- a/src/lib/Gitolite/Hooks/Update.pm	2012-12-19 20:12:31.000000000 +0100
+++ b/src/lib/Gitolite/Hooks/Update.pm	2012-12-19 20:11:52.000000000 +0100
@@ -156,6 +156,10 @@
 __DATA__
 #!/usr/bin/perl
 
+BEGIN {
+    exit 0 if exists $ENV{GL_BYPASS_UPDATE_HOOK};
+}
+
 use strict;
 use warnings;
 
--- a/src/lib/Gitolite/Rc.pm	2012-12-19 20:12:31.000000000 +0100
+++ b/src/lib/Gitolite/Rc.pm	2012-12-19 20:11:52.000000000 +0100
@@ -325,10 +325,10 @@
     # if you're using mirroring, you need a hostname.  This is *one* simple
     # word, not a full domain name.  See documentation if in doubt
     # HOSTNAME                  =>  'darkstar',
-    UMASK                       =>  0077,
+    UMASK                       =>  0007,
 
     # look in the "GIT-CONFIG" section in the README for what to do
-    GIT_CONFIG_KEYS             =>  '',
+    GIT_CONFIG_KEYS             =>  '.*',
 
     # comment out if you don't need all the extra detail in the logfile
     LOG_EXTRA                   =>  1,
